client/jetpack-cloud/sections/partner-portal/issue-license-v2/licenses-form/hooks/use-product-and-plans.tsx

```diff
@@ -12,13 +12,67 @@ import {
 	addSelectedProductSlugs,
 	clearSelectedProductSlugs,
 } from 'calypso/state/partner-portal/products/actions';
+import { APIProductFamilyProduct } from 'calypso/state/partner-portal/types';
+import {
+	PRODUCT_FILTER_ALL,
+	PRODUCT_FILTER_PLANS,
+	PRODUCT_FILTER_PRODUCTS,
+	PRODUCT_FILTER_VAULTPRESS_BACKUP_ADDONS,
+	PRODUCT_FILTER_WOOCOMMERCE_EXTENSIONS,
+} from '../../constants';
 import type { SiteDetails } from '@automattic/data-stores';
 
 type Props = {
 	selectedSite?: SiteDetails | null;
+	selectedProductFilter?: string | null;
+};
+
+const getProductsAndPlansByFilter = (
+	filter: string | null,
+	allProductsAndPlans?: APIProductFamilyProduct[]
+) => {
+	switch ( filter ) {
+		case PRODUCT_FILTER_PRODUCTS:
+			return (
+				allProductsAndPlans?.filter(
+					( { family_slug }: { family_slug: string } ) =>
+						family_slug !== 'jetpack-packs' &&
+						family_slug !== 'jetpack-backup-storage' &&
+						! isWooCommerceProduct( family_slug ) &&
+						! isWpcomHostingProduct( family_slug )
+				) || []
+			);
+		case PRODUCT_FILTER_PLANS:
+			return (
+				allProductsAndPlans?.filter(
+					( { family_slug }: { family_slug: string } ) => family_slug === 'jetpack-packs'
```

Comment:
Nitpick: As the `family_slug` is already defined in the array’s object type, we don't need to declare it again explicitly. 

Instead, we can write:

```
( { family_slug } ) =>
```

Reply:
Thanks for the suggestion. This has been applied 👍